fetch2/src/main/java/com/tonyodev/fetch2/DownloadManager.java

```diff
@@ -250,6 +382,43 @@ public void onPostExecute() {
         actionProcessor.processNext();
     }
 
+    void removeGroup(final String groupId) {
+        if (isDisposed) {
+            return;
+        }
+
+        databaseManager.executeTransaction(new Transaction() {
+
+            @Override
+            public void onPreExecute() {
+            }
+
+            @Override
+            public void onExecute(Database database) {
+                List<RequestData> list = database.queryByGroupId(groupId);
+
+                for (RequestData requestData : list) {
+
+                    if(downloadsMap.containsKey(requestData.getId())){
+                        interrupt(requestData.getId(), InterruptReason.REMOVED);
+
+                    }else {
+                        database.remove(requestData.getId());
```

Comment:
All queries are performed on a background thread and perform relatively fast. However, I have not tested very large queries.